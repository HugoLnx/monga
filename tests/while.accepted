int main() {
  while(x > 10) {
    print("hey! ho! lets go!");
  }
}
-----
int main() {
  while(x > 10)
    print("hey! ho! lets go!");
}
-----
int main() {
  {
    while(x > 10)
      print("hey! ho! lets go!");
  }
}
-----
int main() {
  {
    while(x > 1.0+1e3-2e4+1+0x5) {
      {print("hey! ho! lets go!");}
    }
  }
}
