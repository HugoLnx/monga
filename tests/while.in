int main() {
	int a;
	a = 0;
	
	while (a < 5) {
		printf("%d\n", a);
		a = a + 1;
	}
	a = 0;
	while (a < 10) {
		printf("hey %d\n", a);
		a = a + 1;
	}
}