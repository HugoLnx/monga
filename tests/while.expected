declarations:
  declaration:
    function:main,259,0
      block:
        statements:
          statement:
            while:
              exp:7
                binary:5
                  exp:5
                    var:x
                  exp:0,10
              statement:
                block:
                  statements:
                    statement:
                      attribution:
                        var:y
                        exp:0,20
